packages/core/crmManager/crmManager.ts

```diff
@@ -25,21 +25,24 @@ export default class CrmManager {
     return crmService;
   }
 
-  public async createEvent(event: CalendarEvent) {
+  public async createEvent(event: CalendarEvent, skipContactCreation?: boolean) {
     const crmService = await this.getCrmService(this.credential);
     // First see if the attendees already exist in the crm
     let contacts = (await this.getContacts(event.attendees.map((a) => a.email))) || [];
     // Ensure that all attendees are in the crm
     if (contacts.length == event.attendees.length) {
       return await crmService?.createEvent(event, contacts);
     }
-    // Figure out which contacts to create
-    const contactsToCreate = event.attendees.filter(
-      (attendee) => !contacts.some((contact) => contact.email === attendee.email)
-    );
-    const createdContacts = await this.createContacts(contactsToCreate);
-    contacts = contacts.concat(createdContacts);
-    return await crmService?.createEvent(event, contacts);
+
+    if (!skipContactCreation) {
```

Comment:
NIT: Prefer early returns

```suggestion
    if (skipContactCreation) return
```